src/relax/transform/to_mixed_precision.cc

```diff
@@ -60,36 +60,40 @@ struct NTensorTypeHash {
   }
 };
 
-// Construct a NTensorType from an Expr
-// If the Expr is a Tensor, the returned NTensorType will be the type of the Expr or the given dtype
-// If the Expr is a Tuple, the returned NTensorType will be an array of NTensorType from each field
-NTensorType NTensorTypeFrom(const Expr& expr, DataType dtype = DataType::Void()) {
-  if (const auto* tensor_type = expr->checked_type().as<DynTensorTypeNode>()) {
+// Construct a NTensorType from an Type
+// If the Type is Tensor, the returned NTensorType will be the type or the given dtype
+// If the Type is Tuple, the returned NTensorType will be an array of NTensorType from each field
+NTensorType NTensorTypeFrom(const Type& type, DataType dtype = DataType::Void()) {
+  if (const auto* tensor_type = type.as<DynTensorTypeNode>()) {
```

Comment:
Consider introduce StructInfo instead, since we will reduce the type's usage moving forward in favor of StructInfo